src/storage/btree.c

```diff
@@ -4280,13 +4302,171 @@ btree_read_record (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pgptr, REC
 	}
       else if (n_prefix < 0)
 	{
+	  assert (0);
+	  return n_prefix;
+	}
+    }
+
+  return NO_ERROR;
+}
+
+
+#if !defined(NDEBUG) && defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+static void
+check_validate (BTREE_SCAN * bts)
+{
+  if (bts->is_cur_key_compressed)
+    {
+      assert (bts->common_prefix_size > 0);
+      assert (VPID_ISNULL (&bts->C_vpid) == false);
+      assert (VPID_EQ (&(bts->C_vpid), &bts->cur_common_prefix_page_vpid) == true);
+      assert (bts->C_page != NULL);
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+      assert (VPID_ISNULL (&bts->cur_common_prefix_page_vpid) == false);
+      assert (LSA_EQ (&bts->cur_common_prefix_page_lsa, pgbuf_get_lsa (bts->C_page)));
+#endif
+    }
+}
+#else
+#define check_validate(bts) ;
+#endif
+
+#if defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+static int
+btree_read_record_in_leafpage (THREAD_ENTRY * thread_p, PAGE_PTR pgptr, int copy_key, BTREE_SCAN * bts)
+{
+  int error, n_prefix;
+
+  assert (bts && pgptr && bts->key_record.type == REC_HOME);
+
+  bts->is_cur_key_compressed = false;
+  error =
+    btree_read_record_without_decompression (thread_p, &bts->btid_int, &bts->key_record, &(bts->cur_key),
+					     (void *) &bts->leaf_rec_info, BTREE_LEAF_NODE, &(bts->clear_cur_key),
+					     &bts->offset, copy_key);
+  if (error != NO_ERROR)
+    {
+      return error;
+    }
+
+  assert (btree_leaf_is_flaged (&bts->key_record, BTREE_LEAF_RECORD_FENCE) == false);
+  if (btree_leaf_is_flaged (&bts->key_record, BTREE_LEAF_RECORD_OVERFLOW_KEY)
+      || TP_DOMAIN_TYPE (bts->btid_int.key_type) != DB_TYPE_MIDXKEY)
+    {
+      return NO_ERROR;
+    }
+
+  assert (pgptr == bts->C_page && VPID_EQ (pgbuf_get_vpid_ptr (pgptr), &bts->C_vpid));
+
+  if (bts->common_prefix_size >= 0)
+    {
+      assert (VPID_EQ (&(bts->C_vpid), &bts->cur_common_prefix_page_vpid));
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+      assert (LSA_EQ (&bts->cur_common_prefix_page_lsa, pgbuf_get_lsa (pgptr)));
+      assert (bts->common_prefix_size == btree_node_get_common_prefix (thread_p, &bts->btid_int, pgptr));
+#endif
```

Comment:
How about calling check_validate?

Reply:
Check items are different.
I will keep it as is.  
Thank you.

Reply:
It seems to be the same because it satisfies the following.
```
assert (pgptr == bts->C_page && VPID_EQ (pgbuf_get_vpid_ptr (pgptr), &bts->C_vpid));
```
Compared to check_validate, there is one more line below.
```
assert (bts->common_prefix_size == btree_node_get_common_prefix (thread_p, &bts->btid_int, pgptr));
```
I think this line should be included in check_validate.

Reply:
The purpose is to express my opinion, so if you read it, please resolve it.

Reply:
Overall, I will maintain the current appearance for now.
After reviewing your feedback, some changes were necessary.
Thank you.